### PR TITLE
Upgraded Jinja2 dependency to 3.x

### PR DESCRIPTION
**Pull request by briddo**
### Fixes: #1391

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -97,13 +97,13 @@ graphene-django-optimizer = "~0.8.0"
 importlib-metadata = {version = "~4.4", python = "<3.10"}
 # Template rendering engine
 # NOTE: Jinja2 3.x is available
-Jinja2 = "~2.11.3"
+Jinja2 = "~3.0.3"
 # Optional data validation of config contexts
 jsonschema = "~4.4.0"
 # Rendering of markdown files to HTML
 Markdown = "~3.3.6"
 # Escape text to use HTML and XML. NOTE: later versions are not compatible with Jinja2 2.x
```
Comment (glennmatthews) [inline pyproject.toml:105]: Can remove the NOTE here now.

File: pyproject.toml [hunk 2]
```diff
@@ -97,13 +97,13 @@ graphene-django-optimizer = "~0.8.0"
...
-MarkupSafe = "2.0.1"
+MarkupSafe = "2.1.0"
```
Comment (glennmatthews) [inline pyproject.toml:106]: We don't need to pin to an exact version now - this should be `"~2.1.0"`

**Review by glennmatthews — COMMENTED** [review]
Is there anything we need to test or document about any breaking changes between Jinja2 2.x and 3.x?

**Comment by bryanculver** [comment]
> Is there anything we need to test or document about any breaking changes between Jinja2 2.x and 3.x?

Did a quick glance at their changelogs:
- https://palletsprojects.com/blog/flask-2-0-released/
- https://jinja.palletsprojects.com/en/3.0.x/changes/#version-3-0-0
- https://github.com/pallets/jinja/releases/tag/3.0.0

And it appears they removed code that has been deprecated for some time, most of which seems to be in this commit: https://github.com/pallets/jinja/commit/b0015c72d5acbf93b9d99a1ce6167889338db85b

Where we would have been leveraging that is unclear to me. Nothing else jumps out as a significant breaking change. They also dropped Python 2 and 3.5 support of which a lot of compatibility code was removed for that but should not be felt here.

File: pyproject.toml [hunk 3]
```diff
@@ -36,6 +36,7 @@ python = "^3.7"
 celery = "~5.2.3"
 # Fundamental web framework for Nautobot
 # NOTE: some of our dependencies do not have releases compatible with Django 3.2's DEFAULT_AUTO_FIELD setting
+# but these dependencies now have workarounds in our current code base
```
Comment (bryanculver) [inline pyproject.toml:39]: Is this even necessary anymore? If our dependencies have workarounds for `3.2` doesn't that mean they are compatible?

Comment (briddo) [reply pyproject.toml:39]: Good point!

File: nautobot/docs/release-notes/version-1.3.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ If you are a user migrating from NetBox to Nautobot, please refer to the ["Migra
 
 ## Release Overview
 
+We've upgraded the Jinja2 dependency from version 2.11 to version 3.0.3. This may affect the syntax of any `nautobot.extras.models.ComputedField` objects in your database... Specifically, the `template` attribute, which is parsed as a Jinja2 template. Please refer to [Jinja2 3.0.x's release notes](https://jinja.palletsprojects.com/en/3.0.x/changes/) to check if any changes might be required in your computed fields' templates.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-1.3.md:9]: I don't think this belongs as the first item under the release overview - can you add this as an entry under the release overview `### Changed` heading? instead?

File: nautobot/docs/release-notes/version-1.3.md [hunk 5]
```diff
@@ -71,12 +75,13 @@ As Python 3.6 has reached end-of-life, the default Docker images published for t
...
+- [#1268](https://github.com/nautobot/nautobot/issues/1268) - Drop Support for Python 3.6.
+- [#1277](https://github.com/nautobot/nautobot/issues/1277) - Updated Django dependency to 3.2.X LTS.
 - [#1307](https://github.com/nautobot/nautobot/pull/1307) - Updated various Python package dependencies to their latest compatible versions.
 - [#1314](https://github.com/nautobot/nautobot/pull/1314) - Updated various development-only Python package dependencies to their latest compatible versions.
 - [#1321](https://github.com/nautobot/nautobot/pull/1321) - Updates to various browser package dependencies. This includes updating from Material Design Icons 5.x to 6.x, which has a potential impact on plugins: a [small number of icons have been removed or renamed](https://dev.materialdesignicons.com/upgrade#5.9.55-to-6.1.95) as a result of this change.
 - [#1367](https://github.com/nautobot/nautobot/pull/1367) - Extracted Job-related models to submodule `nautobot.extras.models.jobs`; refined Job testing best practices.
+- [#1391](https://github.com/nautobot/nautobot/pull/1391) - Updated Jinja2 dependency to 3.0.X.
```
Comment (bryanculver) [inline nautobot/docs/release-notes/version-1.3.md:84]: Copypasta I usually make too
```suggestion
- [#1391](https://github.com/nautobot/nautobot/issues/1391) - Updated Jinja2 dependency to 3.0.X.
```

**Review by bryanculver — CHANGES_REQUESTED** [review]

File: nautobot/docs/release-notes/version-1.3.md (hunk 5 above)
Comment (briddo) [reply nautobot/docs/release-notes/version-1.3.md:84]: Because it's already mentioned in the above section?

**Review by bryanculver — APPROVED** [review]

**Pull request merged**